home-manager/home-manager

```diff
@@ -87,14 +86,16 @@ function doBuild() {
 }
 
 function doSwitch() {
-    local wrkdir
-    wrkdir="$(mktemp -d)"
+    local wrkdir="$(mktemp -d)"
 
-    if doBuild "$wrkdir/generation" ; then
-        "$wrkdir/generation/activate"
-    fi
+    local generation
+    local exitCode=0
+    generation=$(doBuild "$wrkdir/result") && $generation/activate || exitCode=1
```

Comment:
Ah, sorry, apparently the `if`-statement will cause the whole script to terminate! Need the `||` as you have it.